sklearn/utils/_openmp_helpers.pyx

```diff
@@ -0,0 +1,38 @@
+IF SKLEARN_OPENMP_SUPPORTED:
+    cimport openmp
+    from . import cpu_count
+
+
+cpdef _openmp_effective_n_threads(n_threads=None):
+    """Determine the effective number of threads used for parallel OpenMP calls
+
+    - For ``n_threads = None``, returns the minimum between the number of cpus
+      and ``openmp.omp_get_max_threads()``.
```

Comment:
This should be rephrased to be more precise: "the number of cpus taking ``cgroups`` quotas into account  and ``openmp.omp_get_max_threads()``. ``cgroups`` quotas can typically be set by tools such as Docker."

Note that ``openmp.omp_get_max_threads()`` already takes number of CPUs (including SMT) and CPU affinity mask constraints into account by default for all OpenMP implementations as far as I know (I have not checked though). So there is no need to mention those here.